test/depen-build-settings/dub.json

```diff
@@ -0,0 +1,8 @@
+{
+	"description": "A minimal D application.",
+	"name": "depen-build-settings",
+
+	"dependencies": {
+		"depend1": { "version" : "*", "dflags" : ["-version=depen_build_settings"] }
```

Comment:
Using `-version` is a pretty bad example, because DUB handles it, and will give you a warning on this.
Use something like `-fPIC` instead, something that needs to be propagated and won't go away.

Reply:
I dont see any warning while building, at which should I receive it? Both `depen1` and `depend2` need `-version=depen_build_settings` to successfully compile. If I remove or change it to something else, compilation fails as expected.

`depen1` depends on `depen2` so, test also checks that the flags are applied transitively to all of the dependency chain.

Reply:
```json
{
    "name": "mytest",
    "dflags": [ "-version=Hello" ]
}
```

Leads to:
```console
$ dub build                                                                                                                                                          :(

## Warning for package mytest ##

The following compiler flags have been specified in the package description
file. They are handled by DUB and direct use in packages is discouraged.
Alternatively, you can set the DFLAGS environment variable to pass custom flags
to the compiler, or use one of the suggestions below:

-version=Hello: Use "versions" to specify version constants in a compiler independent way

Performing "debug" build using dmd for x86_64.
mytest ~master: building configuration "application"...
Linking...
```